Tests/Pcap++Test/Tests/FilterTests.cpp

```diff
@@ -736,4 +735,88 @@ PTF_TEST_CASE(TestPcapFiltersOffline)
 
 	}
 	rawPacketVec.clear();
-}
\ No newline at end of file
+}
+
+PTF_TEST_CASE(TestPcapFilters_matchPacketWithFilter_DifferentInterfaces)
+{
+	// check if matchPacketWithFilter work properly for packets with different LinkLayerType
+
+// pcpp::LINKTYPE_DLT_RAW1 layer
+	pcpp::PcapFileReaderDevice fileReaderDev1(RAW_IP_PCAP_PATH);
+	PTF_ASSERT_TRUE(fileReaderDev1.open());
+	pcpp::RawPacketVector rawPacketVec;
+	fileReaderDev1.getNextPackets(rawPacketVec);
+	fileReaderDev1.close();
+
+	int validCounter = 0;
+	for (pcpp::RawPacketVector::VectorIterator iter = rawPacketVec.begin(); iter != rawPacketVec.end(); iter++)
+	{
+		pcpp::Packet packet(*iter);
+		if(pcpp::IPv4Layer* l = packet.getLayerOfType<pcpp::IPv4Layer>())
```

Comment:
please change the variable name from `l` to something with more meaning like `ip4Layer`

Reply:
Done.